test/util/ChartUtils.spec.tsx

```diff
@@ -242,16 +242,94 @@ describe('parseScale', () => {
 });
 
 describe('getValueByDataKey', () => {
-  const data = { a: 1, b: 2, c: 3 };
+  const data: Record<string, unknown> = {
```

Comment:
Nice, I think this type could be a nice one to generally use for a row of data.

Reply:
Eventually I would like to find a solution that relies on generics instead of unknowns - but for now this will do

Reply:
Also I should really be using `PropertyKey` instead, and adding a test of what happens when DataKey returns a symbol!

Reply:
A side note: I used Symbols in a few workaround to add an index to the data before passing it to recharts. I then use the index to identify an active datapoint from a mouseOver, or a Tooltip. 

So for my use case this would have to be 

```
Record<string | Symbol, unknown>
```

Reply:
Unrelated to this PR, but generally relevant: 

Should we possibly do this in the library itself? The whole getActiveIndex is quite expensive (iterating over the whole dataset every time) and could be done much simpler if we just add an index with a Symbol to the dataset as a preprocessing step in the Chart.